### ISSUE TITLE
Parameter mapping problem in amici_objective

### ISSUE THREAD

**Issue by paulstapor** [ICB-DCM/pyPESTO#664]
**Bug description**
I get a parameter mapping error for a (fairly large) model, when calling the pyPESTO objective function. The model is imported from PEtab. The main problem: A parameter, which is in the parameter_df (and should be estimated), can't be mapped to the amici model. The main error (I think) goes back to
https://github.com/ICB-DCM/pyPESTO/blob/90ed83c3a0064569f02dd37f778bc1f61631ac65/pypesto/objective/amici.py#L345
, which reads
```
    def par_arr_to_dct(self, x: Sequence[float]) -> Dict[str, float]:
        """Create dict from parameter vector."""
        return OrderedDict(zip(self.x_ids, x))
```
The problem here is that x (the nominal parameter vector taken from the petab) is shorter than self.x_ids, i.e., the parameter ids in the pypesto objective (which is an instance of amici_objective). The reason:
`len(x) == sum(petab_problem.parameter_df['estimate'])` (= 10932 in my case)
while 
`len(self.x_ids) == len(petab_problem.parameter_df['estimate']))` (= 10968 in my case)

Hence, it sems like objective creation doesn't care about parameters being fixed in the petab_problem.parameter_df... This sounds super-weird to me. Is is possible that this error escape somehow out tests? Or that nobody found it so far? I'm puzzled... Anyone an intuition whether this is really a pyPESTO problem, or might have some other, weird reason?
Thanks!

**To reproduce**
Not so easy, because I'm not sure how to best boil things down form this model. Particular things: Some parameters in the parameter_df are fixed, but most are to be esimated. Some free parameter_ids are mapped condition-specific via the condition_df (e.g., a general protein decay parameter, which is mapped to cell line specific protein decay parameters, which are to be estimated).

**Environment**
AMICI, lib-petab, and pyPESTO from main branches, ubuntu 20.04, python 3.8 or 3.9 (both)

**Comment by yannikschaelte** [comment]
How did you generate the objective? Note that parameter fixing only happens in `importer.create_problem()(.objective)`, while `importer.create_objective()` doesn't really care. https://github.com/ICB-DCM/pyPESTO/blob/main/pypesto/petab/importer.py#L423

**Comment by yannikschaelte** [comment]
> Ah! Yes, this might be the problem, as my objective was to create the objective, but I need to create the problem first, and then the objective from the problem, which I think was the problem... Let's see...
> 
> Okay, so now seriuosly. If I directly create a pypesto.objective from a petab_problem, this may mess up the fixed parameters, while first creating a pypesto.problem from the petab_problem works? Or, in code:
> 
> ```
> importer = pypesto.petab.PetabImporter(petab_problem)
> pypesto_problem = importer.create_problem()
> pypesto_objective = importer.create_objective()
> ```
> 
> will result in
> `pypesto_objective != pypesto_problem.objective` ?

Objectively, that is correct: When importing an objective to the problem and also passing fixed parameters, the problem takes the objective, clones it and modifies it s.t. it henceforth accepts only reduced input vectors, adding the fixed values itself around that then.

I know that is not super nice, but makes sense if one wants to be able to have both the pure objective, and the parameter-fixed version. For PEtab one could think about having `get_objective` fix 'em anyhow already, as working with the full vector is rather unlikely there.

Directly calling `problem.objective` was never intended, but everyone does that now, so I guess that's fine ...

**Comment by paulstapor** [comment]
Ok, testing it. I guess the overall implementation is fine, I just wasn't aware of this difference, which seems a bit like a subtlety, but can be very important apparently... 

> Directly calling problem.objective was never intended, but everyone does that now, so I guess that's fine ...

Well, you have to hand over an objective if you create a predictor. Would probably then be better to adapt the API to accept a pypesto.problem instead, right?

**Comment by yannikschaelte** [comment]
I think that may be preferable, yes ... I am not sure whether there is a better solution, but this whole handling of fixed parameters always leads to some difficulties, as we initially decided to "know" them inside pypesto (-> in the problem) in order to be able to augment results.

**Comment by paulstapor** [comment]
Okay, then all of this makes sense now, the problem in my code is solved as well. I don't consider this a bug anymore, but a layer 8 problem in my flat... :joy:
Should we document this somehow ind some way? Is this documented? I mean, it's easy to confuse, right? But it's definitely not a bug any more...

**Comment by yannikschaelte** [comment]
I agree that in the petab importer one should more clearly document what happens. When creating an objective from scratch, the problem doesn't occur. Also, whether to fix parameter in get_objective already or not, could also be changed, no preferences.

**Comment by yannikschaelte** [comment]
or maybe to at least provide both options, to fix or not to fix.
